typhos/dynamic_font.py

```diff
@@ -118,26 +126,153 @@ def patch_widget(
         The normalized padding percentage (0.0 - 1.0) to use in determining the
         maximum font size. Content margin settings determine the content
         rectangle, and this padding is applied as a percentage on top of that.
+    max_size : float or None, optional
+        The maximum font point size we're allowed to apply to the widget.
+    min_size : float or None, optional
+        The minimum font point size we're allowed to apply to the widget.
     """
-    def resizeEvent(event: QtGui.QResizeEvent) -> None:
+    if "font-size" in widget.styleSheet():
+        logger.warning(
+            f"Widget named {widget.objectName()} has a fixed size from its "
+            "stylesheet, and cannot be resized dynamically."
+        )
+    if isinstance(widget, QtWidgets.QComboBox):
+        return patch_combo_widget(
+            widget=widget,
+            pad_percent=pad_percent,
+            max_size=max_size,
+            min_size=min_size,
+        )
+    elif hasattr(widget, "setText") and hasattr(widget, "text"):
+        return patch_text_widget(
+            widget=widget,
+            pad_percent=pad_percent,
+            max_size=max_size,
+            min_size=min_size,
+        )
+    else:
+        raise TypeError(f"Dynamic font not supported for {widget}")
+
+
+def patch_text_widget(
+    widget: QtWidgets.QLabel | QtWidgets.QLineEdit,
+    *,
+    pad_percent: float = 0.0,
+    max_size: float | None = None,
+    min_size: float | None = None,
+):
+    """
+    Specific patching for widgets with text() and setText() methods
```

Comment:
This docstring could be a bit more helpful.  I think it'd be nice to get a quick summary of what exactly we're patching: the `setText` and `resizeEvent` methods.  

It seems like this is our entry point to patching other behavior in the future as well, so keeping tabs of exactly what's going on here will be helpful